format-avro/src/main/java/io/cdap/plugin/format/avro/UnstructuredToAvroTransformer.java

```diff
@@ -0,0 +1,85 @@
+
+/*
+ * Copyright © 2018-2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.format.avro;
+
+import com.google.common.collect.Maps;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.plugin.common.RecordConverter;
+import io.cdap.plugin.format.avro.output.GenericRecordWrapper;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.generic.GenericRecordBuilder;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Creates GenericRecords from StructuredRecords without knowing the schema beforehand.
+ */
+public class UnstructuredToAvroTransformer extends RecordConverter<StructuredRecord, GenericRecordWrapper> {
```

Comment:
the StructuredToAvro class allows passing in a null output schema, is this class different?

Reply:
I've removed this class.